sklearn/utils/_array_api.py

```diff
@@ -967,3 +967,20 @@ def _in1d(ar1, ar2, xp, assume_unique=False, invert=False):
         return ret[: ar1.shape[0]]
     else:
         return xp.take(ret, rev_idx, axis=0)
+
+
+def _count_nonzero(X, xp, device, axis=None, sample_weight=None):
+    """A variant of `sklearn.utils.sparsefuncs.count_nonzero` for the Array API."""
+    if axis == -1:
+        axis = 1
+    elif axis == -2:
+        axis = 0
```

Comment:
I think the negative logic was taken directly from the original method that has been defined for sparse arrays.

Reply:
Since the original function seems to be defined for 2d inputs, 
https://github.com/scikit-learn/scikit-learn/blob/210740408a732940430047fe9437c2193735573f/sklearn/utils/sparsefuncs.py#L599-L626
how about updating the docstring and adding the assertion as you suggested?

Reply:
yes, I just took this bit from the `count_nonzero` implementation . I updated now

Reply:
Thanks!